plugins/commands/provision/command.rb

```diff
@@ -27,6 +30,20 @@ def execute
         # Go over each VM and provision!
         @logger.debug("'provision' each target VM...")
         with_target_vms(argv) do |machine|
+          if options[:auto_up] && machine.state.id != :running
+            @logger.debug("current machine state `#{machine.state.id}`. Attempting to bring up.")
+            case machine.state.id
+            when :not_created, :poweroff
+              machine.action(:up)
+            when :paused, :saved
+              machine.action(:resume)
+            else
+              raise Vagrant::Errors::ProvisionAutoUpFailure.new(
+                machine_name: machine.name,
+                machine_state: machine.state.short_description
+              )
+            end
```

Comment:
The only downside I see is that the providers will need to be updated instead of the functionality being intrinsic (more of a concern for non-builtin providers). But given the arbitrary state IDs, it will have to be pushed to the providers. Will give this another consideration and re-implement with a more provider focused implementation.